components/kyma-environment-broker/internal/process/deprovisioning/get_kubeconfig_step.go

```diff
@@ -73,3 +79,7 @@ func (s *GetKubeconfigStep) Run(operation internal.DeprovisioningOperation, log
 
 	return operation, 0, nil
 }
+
+func (s *GetKubeconfigStep) isNotFoundErr(err error) bool {
+	return strings.Contains(err.Error(), "not found")
```

Comment:
nit: is there by any chance an underlying type for the error in provisioner that we can cast on like it's done by apimachinery for kube-api errors?
https://github.com/kubernetes/apimachinery/blob/e3c6631c282ae532562559ff7b2ae93b1687a1ab/pkg/api/errors/errors.go#L525-L536

if there isn't then we probably keep it as is, it's just that I find the type check to be a little bit more robust against changes, e.g. someone can decide the message to become "Not found" or "isn't found" in the future which may break this code.

Reply:
at first I wanted to cast the error returned by provisioner to something identifiable, exactly like it is with k8s client and apimachinery errors, but it is not possible in this case. Provisioner transforms apimachinery errors to its own kind of error: 
https://github.com/kyma-project/control-plane/blob/main/components/provisioner/internal/gardener/provisioner.go#L206
https://github.com/kyma-project/control-plane/blob/main/components/provisioner/internal/util/errors.go#L26
All we get is error message in graphql.

Reply:
Would suggest to use https://pkg.go.dev/errors#Is to check for certain type of errors to support wrapped errors.

Reply:
Since is out of scope to refactor the all other places I think its fine to it this way. But just in general in the codebase the errors are mostly handled unidiomatic.

Reply:
I changed the condition for repeating the step, it shouldn't run for more than 30 mins from the operation creation time, even when we miss "not found" error.